v3/src/models/data/formula-manager.ts

```diff
@@ -191,12 +191,10 @@ export class FormulaManager {
   registerAllFormulas() {
     reaction(() => {
       // Observe all the formulas (their canonical form in fact)
-      const result: string[] = []
+      let result = ""
       this.dataSets.forEach(dataSet => {
         dataSet.attributes.forEach(attr => {
-          if (attr.formula.valid && attr.formula.canonical) {
-            result.push(attr.formula.canonical)
-          }
+          result += `${attr.formula.id}-${attr.formula.canonical}`
```

Comment:
Seems fine for now. Couldn't you also have a computed property that returned an observable map that would potentially allow for more fine-grained changes should the need arise?

Reply:
Is there a need for a returned map to be observable in this case?
I'd worry that it wouldn't work again, as new map would be always different from the old one, as I guess mobx compares references by default. I doubt it tries to be smart and use deep equality, as it didn't do it for a simple array.

The docs only say:
> By default, the result of the data function has to change in order for the effect function to be triggered

But we could have more fine-grained control with the custom `equals` function provided:
https://mobx.js.org/reactions.html#equals-reaction